util/Audio_Transcript_GUI.py

```diff
@@ -0,0 +1,442 @@
+
+            # Find Inference Time
+            s = err.find('Inference took')
+            logging.debug("Inference Time found at: %s" % s)
+            logging.debug(err[s:s+80])
+            inference = err[s:s+80]
+            inference = inference.split()
+            aud_duration = aud_duration + float(inference[4].rstrip('s'))
+            inference_time = inference_time + float(inference[2].rstrip('s.'))
+            logging.debug ("inference_time = %f " %inference_time)
+            logging.debug("aud_duration = %f" %aud_duration)
+            logging.debug("Speech To Text Output: %s" %output.strip())
+            logging.error("Err ? %s" %err.strip())
+
+        # Summary of the files processed
+        f.close()
+
+        # Format pretty
+        title_names = ['Filename', 'Duration', 'Inference Time', 'Model Load Time', 'Remarks']
+        print("%-30s %-20s %-20s %-20s %s" %(title_names[0], title_names[1], title_names[2], title_names[3], title_names[4]))
+        # Extract filename from the full file path
+        filename, ext = os.path.split(os.path.basename(waveFile))
+        print("%-30s %-20s %-20s %-20s" %(filename+ext, aud_duration, inference_time, model_time))
+
+        return "\nTranscription Done..."
+
+
+def read_wave(path):
+    """Reads a .wav file.
+
+    Takes the path, and returns (PCM audio data, sample rate).
+    """
+    with contextlib.closing(wave.open(path, 'rb')) as wf:
+        num_channels = wf.getnchannels()
+        assert num_channels == 1
+        sample_width = wf.getsampwidth()
+        assert sample_width == 2
+        sample_rate = wf.getframerate()
+        assert sample_rate in (8000, 16000, 32000)
+        pcm_data = wf.readframes(wf.getnframes())
+        return pcm_data, sample_rate
+
+
+def write_wave(path, audio, sample_rate):
+    """Writes a .wav file.
+
+    Takes path, PCM audio data, and sample rate.
+    """
+    with contextlib.closing(wave.open(path, 'wb')) as wf:
+        wf.setnchannels(1)
+        wf.setsampwidth(2)
+        wf.setframerate(sample_rate)
+        wf.writeframes(audio)
+
+
+class Frame(object):
+    """Represents a "frame" of audio data."""
+    def __init__(self, bytes, timestamp, duration):
+        self.bytes = bytes
+        self.timestamp = timestamp
+        self.duration = duration
+
+
+def frame_generator(frame_duration_ms, audio, sample_rate):
+    """Generates audio frames from PCM audio data.
+
+    Takes the desired frame duration in milliseconds, the PCM data, and
+    the sample rate.
+
+    Yields Frames of the requested duration.
+    """
+    n = int(sample_rate * (frame_duration_ms / 1000.0) * 2)
+    offset = 0
+    timestamp = 0.0
+    duration = (float(n) / sample_rate) / 2.0
+    while offset + n < len(audio):
+        yield Frame(audio[offset:offset + n], timestamp, duration)
+        timestamp += duration
+        offset += n
+
+
+def vad_collector(sample_rate, frame_duration_ms,
```

Comment:
You already wrote the same code in the other file :)

Reply:
Done ! Moved VAD code to waveSplit.py